tests/test_auto_schema.py

```diff
@@ -0,0 +1,341 @@
+import datetime
+import decimal
+import re
+import typing
+
+import pydantic
+import pytest
+
+import saritasa_sqlalchemy_tools
+
+from . import models, repositories
+
+
+async def test_auto_schema_generation(
+    test_model: models.TestModel,
+) -> None:
+    """Test auto schema generation picks correct types for schema."""
+
+    class AutoSchema(saritasa_sqlalchemy_tools.ModelAutoSchema):
+        class Meta:
+            model = models.TestModel
+            model_config = pydantic.ConfigDict(
+                from_attributes=True,
+                validate_assignment=True,
+            )
+            fields = (
+                "id",
+                "created",
+                "modified",
+                "text",
+                "text_nullable",
+                "text_enum",
+                "text_enum_nullable",
+                "number",
+                "number_nullable",
+                "small_number",
+                "small_number_nullable",
+                "decimal_number",
+                "decimal_number_nullable",
+                "true_false",
+                "true_false_nullable",
+                "text_list",
+                "text_list_nullable",
+                "date_time",
+                "date_time_nullable",
+                "date",
+                "date_nullable",
+                "timedelta",
+                "timedelta_nullable",
+                "related_model_id",
+                "related_model_id_nullable",
+                "custom_property",
+                "custom_property_nullable",
+            )
+
+    schema = AutoSchema.get_schema()
+    model = schema.model_validate(test_model)
+    for field in AutoSchema.Meta.fields:
+        assert getattr(model, field) == getattr(test_model, field)
+        if "nullable" not in field and "property" not in field:
+            with pytest.raises(pydantic.ValidationError):
+                setattr(model, field, None)
+
+
+async def test_auto_schema_type_override_generation(
```

Comment:
For me, it's unclear, what this test checks...
Can you extend docs please?
Or maybe need rename for test